tests/e2e/support/objects/app-files/resource/actions.ts

```diff
@@ -79,9 +81,71 @@ export interface createResourceArgs {
   content?: string
 }
 
-export const createResource = async (args: createResourceArgs): Promise<void> => {
+export const createNewFolder = async ({
+  page,
+  resource
+}: {
+  page: Page
+  resource: string
+}): Promise<void> => {
+  await page.locator(createNewFolderButton).click()
+  await page.locator(resourceNameInput).fill(resource)
+  await Promise.all([
+    page.waitForResponse((resp) => resp.status() === 207 && resp.request().method() === 'PROPFIND'),
+    page.locator(util.format(actionConfirmationButton, 'Create')).click()
+  ])
+}
+
+export const createNewFileOrFolder = async (args: createResourceArgs): Promise<void> => {
+  const { page, name, type, content } = args
+  await page.locator(addNewResourceButton).click()
+  switch (type) {
+    case 'folder': {
+      await createNewFolder({ page, resource: name })
+      break
+    }
+    case 'txtFile': {
+      await page.locator(createNewTxtFileButton).click()
+      await page.locator(resourceNameInput).fill(name)
+      await Promise.all([
+        page.waitForResponse((resp) => resp.status() === 201 && resp.request().method() === 'PUT'),
+        page.locator(util.format(actionConfirmationButton, 'Create')).click()
+      ])
+      await editTextDocument({ page, content: content })
+      break
+    }
+    case 'mdFile': {
+      await page.locator(createNewMdFileButton).click()
+      await page.locator(resourceNameInput).fill(name)
+      await Promise.all([
+        page.waitForResponse((resp) => resp.status() === 201 && resp.request().method() === 'PUT'),
+        page.locator(util.format(actionConfirmationButton, 'Create')).click()
+      ])
+      await editTextDocument({ page, content: content })
+      break
+    }
+    case 'drawioFile': {
+      await page.locator(createNewDrawioFileButton).click()
+      await page.locator(resourceNameInput).fill(name)
+
+      const [drawioTab] = await Promise.all([
+        page.waitForEvent('popup'),
+        page.waitForResponse((resp) => resp.status() === 201 && resp.request().method() === 'PUT'),
+        page.locator(util.format(actionConfirmationButton, 'Create')).click()
+      ])
+      await drawioTab.waitForLoadState()
+      await drawioTab.locator('.geBigButton', { hasText: 'Save' }).isVisible()
```

Comment:
any specific reason why this locator is not in a `const`?

Reply:
will it be possible to do like this?
```js
const bigButton = '.getBigButton'
...
await drawioTab.locator(bigButton, { hasText: 'Save' }).isVisible()
```